packages/playwright/src/matchers/expect.ts

```diff
@@ -104,11 +105,13 @@ export const printReceivedStringContainExpectedResult = (
 
 type ExpectMessage = string | { message?: string };
 
-function createMatchers(actual: unknown, info: ExpectMetaInfo): any {
-  return new Proxy(expectLibrary(actual), new ExpectMetaInfoProxyHandler(info));
+function createMatchers(actual: unknown, info: ExpectMetaInfo, prefix: string[], parentPrefixes: string[][]): any {
```

Comment:
It took me a while to understand why parentPrefixes are needed. I'd suggest a variation of the implementation where instead of parentPrefixes each expect contains a growing list of matchers it has been extended to so far. In that case merge could be implemented like so:

```js
export function mergeExpects(...expects: any[]) {
  let current = expect;
  for (const e of expects)
    current = current.extend(e[getMatchersSymbol]);
  return current;
}
```

Would that work?